modules/libcom/src/osi/os/vxWorks/osdEnv.c

```diff
@@ -35,25 +34,32 @@
  */
 LIBCOM_API void epicsStdCall epicsEnvSet (const char *name, const char *value)
 {
-    char *cp;
+    size_t alen = (!name || !value) ? 2u : strlen(name) + strlen(value) + 2u; /* <NAME> '=' <VALUE> '\0' */
+    const int onstack = alen <= 64u; /* use on-stack dynamic array for small strings */
```

Comment:
I would make the transition point much larger, say 512 bytes, which we would rarely if ever hit. There is enough stack for this kind of thing.

Reply:
Ok, your call.